### PR TITLE
Add endpoint to return available operators

### PR DESCRIPTION
**Pull request by cjh1**

### CODE REVIEW

File: oremda/clients/singularity/client.py [hunk 1]
```diff
@@ -96,3 +98,20 @@ def resolve_path(self, path):
         for p in paths_tried:
             msg += f"  {p}\n"
         raise Exception(msg)
+
+    def images(self, organization: str = None) -> List[ImageBase]:
+        images = []
+
+        images_path = Path(self.images_dir)
+        print(f"path: {images_path}")
```
Comment (psavery) [inline oremda/clients/singularity/client.py:106]: Debug statement?

File: oremda/clients/singularity/client.py [hunk 2]
```diff
@@ -96,3 +98,20 @@ def resolve_path(self, path):
...
+
+        for image_file in images_path.glob(OREMDA_SIF_GLOB_PATTERN):
+            image = SingularityImage(self.client, image_file)
```
Comment (psavery) [inline oremda/clients/singularity/client.py:109]:
```suggestion
            image = self.image(image_file)
```

File: oremda/clients/singularity/image.py [hunk 3]
```diff
@@ -1,10 +1,23 @@
 from oremda.clients.base import ImageBase
+from oremda.constants import SINGULARITY_FROM_LABEL
 
 
 class SingularityImage(ImageBase):
     def __init__(self, client, path):
         self.client = client
         self.path = path
+        self._name = None
+        self._name_extracted = False
```
Comment (psavery) [inline oremda/clients/singularity/image.py:10]: Looks like this is unused?

File: oremda/clients/singularity/client.py (hunk 1 above)
Comment (cjh1) [reply oremda/clients/singularity/client.py]: Yup

File: oremda/clients/docker/client.py [hunk 4]
```diff
@@ -34,3 +36,23 @@ def run(self, *args, **kwargs):
...
+                if organization is not None and not tag.startswith(f"{organization}/"):
+                    continue
+
+                # Check that we have labels
+                docker_image = DockerImage(self.client, tag)
+                labels = docker_image.raw_labels
+
+                # Skip over anything without at least a name
+                if constants.OREMDA_IMAGE_LABEL_NAME not in labels:
+                    continue
```
[the remark this answers is not in the document]
Comment (cjh1) [reply oremda/clients/docker/client.py:54]: Yes, its nice this all worked out!

File: oremda/clients/singularity/client.py (hunk 2 above)
Comment (cjh1) [reply oremda/clients/singularity/client.py]: Thanks, done

File: oremda/clients/singularity/image.py (hunk 3 above)
Comment (cjh1) [reply oremda/clients/singularity/image.py]: Yup, removed

File: oremda/clients/docker/client.py [hunk 5]
```diff
@@ -34,3 +36,23 @@ def run(self, *args, **kwargs):
...
+    def images(self, organization: str = None) -> List[DockerImage]:
+        images = []
+        for image in self.client.images.list():
+            for tag in image.tags:
+                # Apply the organization filter if we have one
+                if organization is not None and not tag.startswith(f"{organization}/"):
+                    continue
+
+                # Check that we have labels
+                docker_image = DockerImage(self.client, tag)
```
[the remark this answers is not in the document]
Comment (cjh1) [reply oremda/clients/docker/client.py]: Done

File: oremda/clients/singularity/client.py [hunk 6]
```diff
@@ -96,3 +98,19 @@ def resolve_path(self, path):
         for p in paths_tried:
             msg += f"  {p}\n"
         raise Exception(msg)
+
+    def images(self, organization: str = None) -> List[ImageBase]:
```
Comment (psavery) [inline oremda/clients/singularity/client.py:102]: The return type here could be `List[SingularityImage]` instead, right? Then I think we also wouldn't need to import `ImageBase`.

Comment (cjh1) [reply oremda/clients/singularity/client.py]: Yup, fixed

**Review by alesgenova — APPROVED** [review]
LETM

**Pull request merged**